### DISCUSSION TITLE
[Bug] `gpt-4-turbo-preview` answer is not up to date.

### DISCUSSION THREAD

**Discussion by RoyRao2333** [lobehub/lobehub#1742] [General Question]
### 💻 Operating System

macOS

### 📦 Environment

Vercel / Zeabur / Sealos

### 🌐 Browser

Chrome

### 🐛 Bug Description

I asked about latest knowledge base and got 2023-04 which is not correct since `gpt-4-0125-preview` is up to date at 2023-12.

<img width="910" alt="image" src="https://github.com/lobehub/lobe-chat/assets/31413093/87a61ac1-8b2e-45dd-ab5c-b7c29eb840b3">

### 🚦 Expected Behavior

Compared to ChatGPT-Next-Web:

<img width="1277" alt="image" src="https://github.com/lobehub/lobe-chat/assets/31413093/d671e8eb-89ea-4ffa-b8bb-cba773353509">

**Comment by zhuozhiyongde** [comment]
This is just a frontend project. Please make sure that the gpt-4-turbo-preview API provided by your provider is correct, and that your configuration of the models parameter is correct.

**Comment by RoyRao2333** [comment]
@zhuozhiyongde 

Yes, I double checked this and I was using the same OpenAI key for both `lobe-chat` and `chatgpt-next-web`.

I also tried to manually specify the exact model name rather than the turbo preview:

<img width="1515" alt="image" src="https://github.com/lobehub/lobe-chat/assets/31413093/d612818f-d668-41ee-bb57-ce1f5cd1da93">

<img width="1240" alt="image" src="https://github.com/lobehub/lobe-chat/assets/31413093/02c95963-bc8c-431b-b18c-d675eb594644">

Still no luck... Any help would be appreaciated!

**Comment by zhuozhiyongde** [comment]
> @zhuozhiyongde
> 
> Yes, I double checked this and I was using the same OpenAI key for both `lobe-chat` and `chatgpt-next-web`.
> 
> I also tried to manually specify the exact model name rather than the turbo preview:
> 
> <img alt="image" width="1515" src="https://private-user-images.githubusercontent.com/31413093/316514459-d612818f-d668-41ee-bb57-ce1f5cd1da93.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTEzNzc2NTQsIm5iZiI6MTcxMTM3NzM1NCwicGF0aCI6Ii8zMTQxMzA5My8zMTY1MTQ0NTktZDYxMjgxOGYtZDY2OC00MWVlLWJiNTctY2UxZjVjZDFkYTkzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzI1VDE0MzU1NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY5Yzc1MWNkNzAyNDNmNjk2N2NiMDljMDk0NmY5NzliODE1NTRlZTUxODk4YzQ0MGM0NjllZDA4MmNjMjg2NTcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.pdx2jQP-bGzIaDr3kX3po67H9R7H9fCQxyeuULgq5pg"> <img alt="image" width="1240" src="https://private-user-images.githubusercontent.com/31413093/316514112-02c95963-bc8c-431b-b18c-d675eb594644.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTEzNzc2NTQsIm5iZiI6MTcxMTM3NzM1NCwicGF0aCI6Ii8zMTQxMzA5My8zMTY1MTQxMTItMDJjOTU5NjMtYmM4Yy00MzFiLWIxOGMtZDY3NWViNTk0NjQ0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzI1VDE0MzU1NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTc0ZWQ5NmIwZDgwNWEyMGRkMzhhNmRhMzQxZWM3MTMyZjMxYmVhZmZkOTRkNzM5ZWRkN2RjZTM5NmQxNGZiNmQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.TjY3YgmLbQw88CHHprP2hJnIjVUeLn5IABIFZCZV42U">
> Still no luck... Any help would be appreaciated!

Maybe this is an issue with OpenAI official. This is the result of me directly requesting the official API using Hoppscotch:

![image](https://github.com/lobehub/lobe-chat/assets/36695271/077f12b2-5e1d-48e2-8b94-eb06a1f14906)

**Comment by zhuozhiyongde** [comment]
![image](https://github.com/lobehub/lobe-chat/assets/36695271/5033cff7-e89e-4c9a-a47e-61b1bdce1d3d)

[a comment by an automated account is not shown]

**Comment by RoyRao2333** [comment]
@zhuozhiyongde 

Maybe you're right. Don't know why 'chathpt-next-web' has different results, but thanks anyway. 

I think I'll wait some time till it's fixed by OpenAI.

**Comment by RoyRao2333** [comment]
@zhuozhiyongde 

大概猜测了一下问题：因为`lobe-chat`请求时需要中转，而我使用`chatgpt-next-web`的时候是前端填写OpenAI Key，而非部署时的环境变量，所以`chatgpt-next-web`是前端直接请求（根据以下discussion所讨论的，`lobe-chat`前端填写key貌似也会中转）。而不同模型所支持的地区也不同，可能经过中转后的请求区域不在支持范围内，所以本项目实际使用的模型和`chatgpt-next-web`不一样。

https://github.com/lobehub/lobe-chat/discussions/728#discussioncomment-7898361

<img width="803" alt="image" src="https://github.com/lobehub/lobe-chat/assets/31413093/3c4ea5f9-b98b-4b69-b97c-0fe98148ce6d">

`lobe-chat`的api：`vercel.app/api/chat/openai`

`chatgpt-next-web`的api：`vercel.app/api/openai/v1/chat/completions`

[a comment by an automated account is not shown]

**Comment by arvinxx** [comment]
和请求没关系。单纯是因为 Chat-Next-Web 加了一层扮演 ChatGPT 的 system Role prompt:

https://github.com/ChatGPTNextWeb/ChatGPT-Next-Web/blob/133ce39a13cb90733bc0aac220ea179e34fd4430/app/constant.ts#L97-L118

而我觉得这东西没啥用，没加而已。

**Reply by RoyRao2333** [reply]
虽然没有研究过`chatgpt-next-web`的代码，但从你发的选段来看，是他们自己维护了一个常量来回复截至日期的？😂

**Reply by MapleEve** [reply]
是的，就是这么糊弄，反正用的人也不看代码

**Reply by RoyRao2333** [reply]
好的，感谢各位大佬了 🤣

[a comment by an automated account is not shown]
